### PR TITLE
created makefile to for linting, testing, formatting

### PR DESCRIPTION
**Pull request by ericmjl**
cc @eigenfoo @canyon289 

This refactors the scripts into makefile commands. 

Need help getting the commands right for the docker containers in .travis.yml, though.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+
+style:
+	echo "Checking code style with pylint..."
+	python -m pylint pymc4/
+	echo "Success!"
+
+
+test:
+	pytest -v pymc4/tests/ --cov=pymc4/
```
Comment (canyon289) [inline Makefile]: Would be nice to add html output coverage output here for non computer users

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,25 @@
+docstyle:
```
Comment (canyon289) [inline Makefile]: Do a `phony` target need to be added to all of these?

https://www.gnu.org/software/make/manual/html_node/Phony-Targets.html

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+	echo "Success!"
+
+
+test:
+	pytest -v pymc4/tests/ --cov=pymc4/
+
+lint: docstyle format style
+
+
+check: lint test
```
Comment (canyon289) [inline Makefile]: Deleting `lint.sh` is going to be break CI. `.travisyml` file will need to be updated as well

Comment (ericmjl) [reply Makefile]: Yep, exactly what I expected. I have identified the lines that I need to modify:

```
  - if [[ $NAME == UNIT ]]; then docker run  --mount type=bind,source="$(pwd)",target=/opt/pymc4 --name pymc4_test pymc4:latest bash -c \
                                              "pytest -v pymc4/tests/ --cov=pymc4/";
    fi

  - if [[ $NAME == LINT ]]; then docker run pymc4:latest bash -c "./scripts/lint.sh";
    fi
```

How would you modify these? I'm not quite well-versed with Docker.

Comment (canyon289) [reply Makefile]: >   - if [[ $NAME == LINT ]]; then docker run pymc4:latest bash -c "make lint";
>     fi

That should work. You can leave the other one for now it'll still function

File: Makefile (hunk 2 above)
Comment (ericmjl) [reply Makefile]: I don't think so - regardless of whether files on disk have changed or not, make check works reliably.

File: Makefile (hunk 1 above)
Comment (ericmjl) [reply Makefile]: Do you have any opinions/prior knowledge on the best directory to store the output? 

We can generate a "testing-report.html", and add it to gitignore so it doesn't accidentally get checked into source control, perhaps.

Comment (canyon289) [reply Makefile]: The output isn't just one html file, I think its a whole directory of html files. Adding it to `.gitignore`. is a great idea

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,25 @@
+docstyle:
+	echo "Checking documentation with pydocstyle..."
+	python -m pydocstyle ${SRC_DIR}/pymc4/
+	echo "Success!"
+
+
+format:
+	echo "Checking code style with black..."
+	python -m black -l 100 --check pymc4/
```
Comment (eigenfoo) [inline Makefile]: Among the CI tools we use to check code, black is the only one that offers to in-place format your code for you. I'd like to take advantage of that: what do you think of having `make black` run `black -l 100 pymc4/` (without the `--check` flag)?

Comment (canyon289) [reply Makefile]: We would need both a check and a format make command if we do this. That way the CI engine can check for black formatting without modifying the code

Comment (eigenfoo) [reply Makefile]: Right, currently all recipes only check and don't modify code: I'm suggesting a new recipe (`make black` perhaps?) to do what I'm describing.

Comment (canyon289) [reply Makefile]: We also could simplify this to

`black --check pymc4/`

File: Makefile [hunk 5]
```diff
@@ -17,7 +17,7 @@ style:
 
 
 test:
-	pytest -v pymc4/tests/ --cov=pymc4/
+	pytest -v pymc4/tests/ --cov=pymc4/ --html=testing-report.html --self-contained-html
```
Comment (canyon289) [inline Makefile]: This is cool!

File: Makefile (hunk 4 above)
Comment (ericmjl) [reply Makefile]: See commit: DA MASTAH BOMB FOMATTAH

File: Makefile (hunk 3 above)
Comment (ericmjl) [reply Makefile]: Solved.

File: Makefile (hunk 1 above)
Comment (ericmjl) [reply Makefile]: We're good - I figured out how to make it self-contained, with no assets directory necessary.

File: Makefile (hunk 4 above)
Comment (ericmjl) [reply Makefile]: Added a "black" command, and "check" now just runs black to format the code. "Check" is designed for a pre-commit check.

File: Makefile (hunk 5 above)
Comment (ericmjl) [reply Makefile]: Yay!

**Comment by ericmjl** [comment]
@canyon289 @eigenfoo do let me know what else you think needs to be changed :smile:.

**Comment by canyon289** [comment]
I'm just waiting for CI

**Comment by eigenfoo** [comment]
Looks good, thanks @ericmjl!

File: Makefile (hunk 4 above)
Comment (canyon289) [reply Makefile]: Forgot to simplify this but nothing really changes ¯\_(ツ)_/¯

Was going to check this when I rereviewed but review got dismissed

**Pull request merged**